internal/analytics/history_test.go

```
@@ -0,0 +1,451 @@
+	}
+
+	// VACUUM INTO a controlled path (bypasses same-second filename collision).
+	dest := filepath.Join(snapDir, name+".db")
+	quoted := "'" + escapeSQLLiteral(dest) + "'"
+	_, err := db.db.ExecContext(ctx, "VACUUM INTO "+quoted)
+	require.NoError(t, err)
+	return dest
+}
+
+// copyFile copies src to dst, creating dst if needed.
+func copyFile(t *testing.T, src, dst string) {
```

golangci-lint (2.12.2)

[error] func copyFile is unused

(unused)